barebox: readd lzop-native

Emantor opened the pull request:
In scarthgap, lzop has been readded to poky with commit 07006c486722 ("Revert "lzop: remove recipe from oe-core""), readd it to the barebox depends to reallow lzop compression.

This reverts commit 38ada68c708d793eae99ace05dae1ba061fbfbdc.

ejoerns commented:
@Emantor Do we really still need it for newer barebox integrations?

a3f commented:
The upstream configs have been switched to LZ4 in https://lore.barebox.org/barebox/20221206112614.2612071-1-m.felsch@pengutronix.de/#t, but LZO remains a better compression choice if available. ZSTD would be a nice replacement for LZO, but the decompressor turned out to be quite heavy: https://lore.barebox.org/barebox/20220713100922.1880282-1-a.fatoum@pengutronix.de/

Emantor commented:
> @Emantor Do we really still need it for newer barebox integrations?

Regardless whether we have switched over integrations since the removal, LZO is currently supported within upstream barebox and as the poky commit notes still the best time/compression compromise on lower end ARM systems. I don't see a reason to not support LZO.

ejoerns commented:
> > @Emantor Do we really still need it for newer barebox integrations?
> 
> Regardless whether we have switched over integrations since the removal, LZO is currently supported within upstream barebox and as the poky commit notes still the best time/compression compromise on lower end ARM systems. I don't see a reason to not support LZO.

It's not an option to add the dependency for cases where it's actually used?

Also, note that the recipe is a bit deprecated anyway.

nit: the referenced commit is a poky commit, not an oe-core commit

Emantor commented:
> > > @Emantor Do we really still need it for newer barebox integrations?
> > 
> > 
> > Regardless whether we have switched over integrations since the removal, LZO is currently supported within upstream barebox and as the poky commit notes still the best time/compression compromise on lower end ARM systems. I don't see a reason to not support LZO.
> 
> It's not an option to add the dependency for cases where it's actually used?

Yes, I already did that. From a usability perspective I want to readd this to the upstream recipe since especially older kirkstone BSPs are prone to have this set which will stumble upon this during a scarthgap branch. Since LZO is supported both in poky/oe-core and barebox, I don't see a reason to fix this for every BSP. 

> Also, note that the recipe is a bit deprecated anyway.

In that case we should add a deprecation note and have a removal plan.

> nit: the referenced commit is a poky commit, not an oe-core commit

Fixed this with a force push.
